app/services/geo_discovery/document_builder/spatial_builder.rb

```diff
@@ -19,7 +19,10 @@ def build(document)
         # Parses coverage field from geo resource and instantiates a coverage object.
         # @return [GeoCoverage] coverage object
         def coverage
-          @coverage ||= GeoCoverage.parse(resource_decorator.coverage.first)
+          @coverage ||= begin
+                          valid_coverages = resource_decorator.coverage.map { |c| GeoCoverage.parse(c) }.compact
```

Comment:
Do we want to push this logic to the decorator?

Reply:
In this case, the decorator is special decorator that merges direct and imported metadata. I'd rather not clutter that with one-off business logic.

https://github.com/pulibrary/figgy/blob/master/app/decorators/geoblacklight_metadata_decorator.rb